### DISCUSSION TITLE
Blog functionality in Decouple CMS

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#12437] [Q&A]
Require how to implement end to end blog functionality in an existing decouple CMS with the help of content type or any recommended solution, please advise folks @Skrypt @ns8482e

**Comment by hishamco** [comment]
You can checkout TheBlogTheme for how blog is implemented, also you can check our docs for Decouple CMS

**Reply by saleemhaq** [reply]
> You can checkout TheBlogTheme for how blog is implemented, also you can check our docs for Decouple CMS

how can we implemented in an existing theme, we create a decouple cms that have theme inside that theme means withing same theme i need to implement that functionality we have a menu inside the menu have option blog where we have there custom blog page and in seprate blog we have read more button once user click that button he land to detail page where he also comment on that blog also.

**Reply by hishamco** [reply]
As I said before, please spend some time check the blog theme and how we implement such functionality

**Comment by ns8482e** [comment]
@saleemhaq IMHO decoupled cms means, you create data(content) using CMS system(in OC ie admin), However frontend is what you create and isnot dependent (not coupled)on OC themes and shapes instead you build your own UI using data(content) created

So in decouple you only use OC’s features like IOrchardHelper. content management, content api, graphql  and queries for data access

everything else for UI you develop on your own

this tutorial is good start for decoupled site https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/#creating-a-new-decoupled-cms-website

Due to nature of  decoupled there can’t be end to end solution - as every decoupled website will be unique to their needs

**Reply by saleemhaq** [reply]
> @saleemhaq IMHO decoupled cms means, you create data(content) using CMS system(in OC ie admin), However frontend is what you create and isnot dependent (not coupled)on OC themes and shapes instead you build your own UI using data(content) created
> 
> So in decouple you only use OC’s features like IOrchardHelper. content management, content api, graphql and queries for data access
> 
> everything else for UI you develop on your own
> 
> this tutorial is good start for decoupled site https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/#creating-a-new-decoupled-cms-website
> 
> Due to nature of decoupled there can’t be end to end solution - as every decoupled website will be unique to their needs

Can you share with any example or sample source for it really helpful for me for implementing purpose end to end. @ns8482e thanks in advnace. @Skrypt

**Reply by ns8482e** [reply]
I don’t understand what you mean by end to end, decoupled means you can build as complex website as you want using asp.net core like any other website, except you read data from orchard

**Reply by ns8482e** [reply]
There are different ways to read data in orchard, easiest way to use orchard helper

**Reply by saleemhaq** [reply]
> Can you share with any example or sample source for it really helpful for me

Can you share with any example or sample source for it really helpful for me. I need to use this in decouple CMS

**Reply by saleemhaq** [reply]
> > Can you share with any example or sample source for it really helpful for me
> 
> Can you share with any example or sample source for it really helpful for me. I need to use this in decouple CMS

The blog theme is in liquid

![image](https://user-images.githubusercontent.com/5202291/191582034-d9174c92-cf7c-4ed9-b909-8c7ccadbb50e.png)

 we have in razor syntax also if have please share for understaning my understanding in razor please advise @ns8482e @Skrypt
